New branch

Spolpinka opened the pull request:
все эндпойнты реализованы, тестированы

Spolpinka committed:
че то пока ниче не работает

Spolpinka committed:
ага, сохранение заработало

Spolpinka committed:
не может распарсить файл базы данных

Spolpinka committed:
работает, только надо еще статичный ID сохранять

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.) еще без теста

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.) еще без теста

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.)
+ вывод всех транзакций

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.)
+ вывод всех транзакций

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.)
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.)
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков

Spolpinka committed:
работает, только надо еще статичный ID сохранять
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков
+ выгрузка базы в Json

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков
+ выгрузка базы в Json
+ импорт всей базы

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков
+ выгрузка базы в Json
+ импорт всей базы

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков
+ выгрузка базы в Json
+ импорт всей базы

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/controllers/SocksController.java:
```diff
@@ -0,0 +1,293 @@
+            }
+
+    )
+    @PutMapping(value = "/addSocks")
```
Добавление это обычно POST, изменение чего-то существующего это PUT

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/controllers/SocksController.java:
```diff
@@ -0,0 +1,293 @@
+        }
+    }
+
+    @DeleteMapping("/delete")
```
`/detele` можно не писать, будет понятно по типу запроса

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/controllers/SocksController.java:
```diff
@@ -0,0 +1,293 @@
+        }
+    }
+
+    @GetMapping("/concreteSocks")
```
Тут в принципе тоже путь можно не указывать, спринг сможет различить какой метод вызывать по аргументам метода

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/controllers/SocksController.java:
```diff
@@ -0,0 +1,293 @@
+            }
+
+    )
+    @PostMapping(value = "/pickUpSocks", consumes = MediaType.APPLICATION_JSON_VALUE)
```
А вот тут как раз должен быть PUT

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/services/impls/SocksServiceImpl.java:
```diff
@@ -0,0 +1,246 @@
+        readFromFile();
+    }
+
+    //добавление. Вопрос: именно здесь прога увидит, что в массиве нет объектов sock? И значит ли это, что массив null?
```
Не совсем понял, что значит нет объектов sock? Нужные проверки лучше сделать в контроллере, чтобы сюда приходил либо пустой массив, либо заполненный чем-то. Чтобы можно было сразу цикл написать без всяких проверок

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/services/impls/SocksServiceImpl.java:
```diff
@@ -0,0 +1,246 @@
+
+    @Override
+    public String pickUpSocks(Sock[] takenSocks) {
+        if (takenSocks != null) {
```
Аналогичный момент с проверкой, лучше перенести ее в контроллер чтобы тут этого не делать. Не придется возвращать false как строку.

alekseykoshelev commented on src/main/java/pro/sky/coursework/courseworksocks/services/impls/SocksServiceImpl.java:
```diff
@@ -0,0 +1,246 @@
+                        System.out.println("Транзакция не прошла на добавление!");
+                    }
+                } else {
+                    isEnough = false;
```
Думаю, лучше вместо этого флажка использовать return. Придется переписать код, но как мне кажется, он станет красивее)

Spolpinka committed:
работает, ID сохраняется
+ транзакции (сохранение, загрузка, и т.п.) поправил сохранение
+ вывод всех транзакций
+ оттестированы добавление и забор носков
+ добавлены комменты к swagger-у
+ добавлен эндпойнт списания носков
+ выгрузка базы в Json
+ импорт всей базы
+ вроде всё
